plugins/ruby/codegen.pl

```diff
@@ -534,6 +539,18 @@ sub render_class_vmethod_ret {
         };
         push @lines_rb, "end._at(ptr) if ptr != 0";
     }
+    elsif ($retmeta eq 'primitive')
```

Comment:
is this condition inverted?

Reply:
is stl-string a primitive? I would assume no, but I might be wrong.

Reply:
I didn't think stl-strings were considered primitives in df-structures, at least, but I could be wrong about that.

Reply:
lower-1.xslt defines stl-string and stl-fstream to be primitive

Reply:
Oops, I was looking at the diff and didn't see your other comment (below here).